### PR TITLE
fix: issues/336: typescript: importing component from pnpm monorepo p…

### PR DESCRIPTION
**Pull request by btakita**
fixes https://github.com/sveltejs/svelte-preprocess/issues/336

### CODE REVIEW

**Review by kaisermann — APPROVED** [review]

**Comment by ashpieboop** [comment]
In my setup, some files are passed with relative path, no leading `./` (example: `build/views/home_dep.svelte`) so I believe your fix is incomplete.

File: src/transformers/typescript.ts [hunk 1]
```diff
@@ -144,7 +144,10 @@ const transformer: Transformer<Options.Typescript> = ({
     sourceMapText: map,
     diagnostics,
   } = ts.transpileModule(content, {
-    fileName: filename,
+    fileName:
+      filename.slice(0, 1) === '.'
```
Comment (ashpieboop) [inline src/transformers/typescript.ts:148]: replace with `!isAbsolute(filename)` (from `path`)

**Review by ashpieboop — CHANGES_REQUESTED** [review]

File: src/transformers/typescript.ts [hunk 2]
```diff
@@ -144,7 +144,10 @@ const transformer: Transformer<Options.Typescript> = ({
...
+        ? resolve(join(process.cwd(), filename))
```
Comment (ashpieboop) [inline src/transformers/typescript.ts:149]: replace with `? resolve(join(basepath, filename))`

**Review by ashpieboop — CHANGES_REQUESTED** [review]

File: src/transformers/typescript.ts (hunk 1 above)
Comment (btakita) [reply src/transformers/typescript.ts]: Applied in https://github.com/sveltejs/svelte-preprocess/commit/842867034f90c12d6a5fc523477133c8117b5d38. Thank you!

File: src/transformers/typescript.ts (hunk 2 above)
Comment (btakita) [reply src/transformers/typescript.ts]: Ditto. https://github.com/sveltejs/svelte-preprocess/commit/842867034f90c12d6a5fc523477133c8117b5d38

**Comment by kaisermann** [comment]
@btakita Thanks for the contribution! However, I think that the last rebase/merge done added quite a few of undesired commits to the commit history of this branch 🤔

**Comment by btakita** [comment]
@kaisermann np, I will remove them.

**Comment by kaisermann** [comment]
@btakita There were still some extra commits there. I took the liberty of rewriting the branch's commit history. Could you check if the code is still as it should? If yes, we're good to go 👍

**Comment by btakita** [comment]
@kaisermann lgtm. Thank you for removing those commits.

**Review by ashpieboop — APPROVED** [review]

**Comment by kaisermann** [comment]
Released in 4.7.3 🎉

**Pull request merged**